Don't cut the value on `change`

btecu opened the pull request:
Android does not support `keypress` which means when `change` runs, it will use the input and set the placeholders.
It doesn't solve the Android problem but it makes the input field usable.

In addition, cutting the value will erroneously remove an additional character:
Have a mask such as `111-11-1111` and enter `123-45-6789`. If you remove the `6` and leave the field, the cut will remove the ending character and will cause the input to be `123-45-_78_`. With the new change it will be `123-45-789_`.

btecu commented:
Travis failing is not related.

hlubek commented:
Which browser (and version) on Android did you test?

This breaks support for cut in other browsers. I compared it with the "original" React version and it's the same code as we use.

Can we find a way to support both Android and cut on other browsers?

btecu commented:
Try any Samsung phone. The browser is Chrome.

I haven't tried cut, but I think being able to enter anything is more important than being able to cut.

btecu committed:
Don't cut the value on `change`

Android does not support `keypress` which means when `change` runs, it will use the input and set the placeholders.
It doesn't solve the Android problem but it makes the input field usable.

In addition, cutting the value will erroneously remove an additional character:
Have a mask such as `111-11-1111` and enter `123-45-6789`. If you remove the `6` and leave the field, the cut will remove the ending character and will cause the input to be `123-45-_78_`. With the new change it will be `123-45-789_`.

btecu committed:
The placeholder should be passed in when needed

btecu commented:
The issue still exists with Android but I no longer use the library.
